### ISSUE TITLE
[ENHANCEMENT] Smarter Notifications in Windows Action Center

### ISSUE THREAD

**Issue by timothymowens** [marticliment/UniGetUI#2061]
### Please confirm these before moving forward

- [X] I have searched for my feature proposal and have not found a work-in-progress/duplicate/resolved/discarded issue.
- [X] This improvement refers to an existing feature. If you want to suggest a new feature, please use [this template](https://github.com/marticliment/WingetUI/issues/new?assignees=marticliment&labels=new-feature&projects=&template=feature-request.yml&title=%5BFEATURE+REQUEST%5D+%28Enter+your+description+here%29)
- [X] This improvement is not a bug. If you want to report a bug, please use [this template](https://github.com/marticliment/WingetUI/issues/new?assignees=marticliment&labels=bug&projects=&template=bug-issue.yml&title=%5BBUG%5D+%28Enter+your+description+here%29)

### Describe the improvement

Currently WingetUI will display a notification in the Windows Action Center when there are updates available, when updates fail, etc.  I propose making this a bit smarter so that the Action Center isn't so flooded with notifications.  Applications have the ability to remove their "toast" notifications from the Action Center just the same as they can create them.

For some notifications, there should only be a maximum of 1 displayed in Action Center, such as in the following scenarios:

1. When WingetUI checks for and finds updates, it will create a toast notification in the Action Center.  When it does this again later, it will again create another toast notification in the Action Center.  However, before actually creating the new one, the application should do a quick check to see if another notification with a similar message already exists and if so, remove the existing one before creating the new one.  In this way, there will only be a maximum of 1 update notification in the Action Center.
2. When an app update or install fails, WingetUI creates a toast notification, but when the install/update is retried and fails or succeeds, the older fail notification still exists in Action Center, but now another notification for the same application will be created.  Again, in this scenario, WingetUI should check for and delete the older notification if one exists before creating the new one.

Basically the request is to be a little bit smarter with Action Center notifications so as to not flood it with duplicates so much.

### Describe how this improvement could help users.

Only the most recent and relevant notifications will exist in the Windows Action Center related to WingetUI.  Only the most recent update notification or app install/update success or failure notification from WingetUI, etc.  There will be less notification clutter caused by WingetUI.

**Comment by marticliment** [comment]
> * When WingetUI checks for and finds updates, it will create a toast notification in the Action Center.  When it does this again later, it will again create another toast notification in the Action Center.  However, before actually creating the new one, the application should do a quick check to see if another notification with a similar message already exists and if so, remove the existing one before creating the new one.  In this way, there will only be a maximum of 1 update notification in the Action Center.

This behaviuor should have been fixed in 3.0.1, i suppose it is not working, right?

**Comment by timothymowens** [comment]
This is the most recent version of WingetUI and it's still occurring for me:

[image: image.png]

**Comment by timothymowens** [comment]
Apologies, here's the image:

![image](https://github.com/marticliment/WingetUI/assets/3083890/11dce626-def8-414a-b002-fbcc5397dbea)

**Comment by marticliment** [comment]
Thanks!

Will investigate this

**Comment by kriswilk** [comment]
Just wanted to report that with the latest UniGetUI (3.1.0) I'm also seeing repeated notifications in the Action Center:

![image](https://github.com/marticliment/UniGetUI/assets/382436/34a1da33-7417-493e-9f3c-df475db71c87)

All 16 of those notifications were generated overnight, all for the same package update. This is particularly annoying when I wake my PC in the morning and get a huge flood of these.

Personally, I would prefer to be notified ONCE ONLY when each update is found, not periodically. If a notification has already been sent to the Action Center for an update and I've dismissed it, I really don't want to get it again (and again, and again).
